src/plugins/data/public/search/aggs/buckets/terms.ts

```diff
@@ -60,6 +60,21 @@ export interface TermsBucketAggDependencies {
   getInternalStartServices: GetInternalStartServicesFn;
 }
 
+export interface AggParamsTerms {
```

Comment:
It's not clear enough why you created that interface in `buckets/terms.ts` not in `buckets/terms_fn.ts`
From my point of view we can simplify your approach and remove `AggExpressionFunctionArgs` and `AggParamsMapping` types at all. I don't see any benefits for today.

what about: 

```
import { AggExpressionType, AggConfigJson, IAggExpressionFunction } from '../';

const aggTypeName = 'terms';
const fnName = 'aggTypeTerms';

type Input = any;
type Output = AggExpressionType;

export interface AggParamsTerms extends IAggExpressionFunction {
  field: string;
  order: 'asc' | 'desc';
  orderBy: string;
  orderAgg?: AggConfigJson;
  size?: number;
  missingBucket?: boolean;
  missingBucketLabel?: string;
  otherBucket?: boolean;
  otherBucketLabel?: string;
  // advanced
  exclude?: string;
  include?: string;
}

type FunctionDefinition = ExpressionFunctionDefinition<
  typeof fnName,
  Input,
  AggParamsTerms,
  Output
>;

export const aggTypeTerms = (): FunctionDefinition => ({

```

Reply:
Part of the reason I put this with `terms.ts` & created the `AggParamsMapping` was because I felt that having typings for each of the agg type params would be generally useful outside of the expression functions & didn't have anything to do with expressions in particular.

For example, say I want to create a new agg config. Without the mapping, I could do:
```ts
const configState = {
  type: 'terms',
  enabled: true,
  params: {
    field: 'machine.os.keyword',
    // ...etc
  }
};
data.search.aggs.createAggConfigs(indexPattern, [configState]);
```

But `configState` is only enforced to match the general shape of params (currently pretty useless: `Record<string, any>`). What would be ideal would be to have the ability to pull in an interface for the params you are using so that you can have more type safety in your params. Perhaps by making the serialized agg config interface a generic:

```ts
const configState: AggConfigSerialized<AggParamsTerms> = {
  type: 'terms',
  enabled: true,
  params: {
    field: 'machine.os.keyword',
    // ...now this would be more strictly typed
  }
};
```

Or even better, using the type from the global `AggParamsMapping` based on the type name string.

More concretely, as we have an agg types registry which technically could have anything registered to it, having a mapping of these interfaces that others can add to via `declare module` will be necessary eventually anyway.